File: src/main/java/rlbot/render/NamedRenderer.java
package rlbot.render;

import com.google.flatbuffers.FlatBufferBuilder;
import rlbot.cppinterop.RLBotDll;

/**
 * This renderer must be constructed with a specific unique name. You can draw, clear, etc. independently from
 * other renderers.
 */
public class NamedRenderer extends Renderer {
    public NamedRenderer(final String renderName) {
        super(renderName.hashCode());
    }

    public void startPacket() {
        builder = new FlatBufferBuilder(1000);
    }

    public RenderPacket finishPacket() {
        return doFinishPacket();
    }

    public void finishAndSend() {
        RLBotDll.sendRenderPacket(finishPacket());
    }
}

File: src/main/cs/RLBotDotNet/RLBotDotNet/Renderer/NamedRenderer.cs
using FlatBuffers;

namespace RLBotDotNet.Renderer
{
    /// <summary>
    /// Renderer that uses a name as its index.
    /// </summary>
    public class NamedRenderer : Renderer
    {
        /// <summary>
        /// Constructs a new instance of NamedRenderer using the render name as the index.
        /// </summary>
        /// <param name="renderName">The name to use as the render index.</param>
        public NamedRenderer(string renderName) : base(renderName.GetHashCode()) { }

        /// <summary>
        /// Starts the render packet.
        /// </summary>
        public void StartPacket()
        {
            Builder = new FlatBufferBuilder(1000);
        }

        /// <summary>
        /// Finishes and sends the render packet.
        /// </summary>
        public void FinishAndSend()
        {
            SendPacket(FinishPacket());
        }

        private RenderPacket FinishPacket()
        {
            return DoFinishPacket();
        }
    }
}